### PR TITLE
Minors updates founds along the way of debuging kakarot in madara/katana

### PR DESCRIPTION
**Pull request by ClementWalter**
Time spent on this PR: 1 day

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

No way to easily extract all the hints used.

## What is the new behavior?

Added a (chat GPT generated) script

### CODE REVIEW

File: scripts/utils/starknet.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def compile_contract(contract):
             BUILD_DIR / f"{contract['contract_name']}.json",
             "--cairo_path",
             str(SOURCE_DIR),
-            "--no_debug_info",
+            *(["--no_debug_info"] if NETWORK != "devnet" else []),
```
Comment (Eikix) [inline scripts/utils/starknet.py:257]: can we add "katana" as well

Comment (Eikix) [reply scripts/utils/starknet.py:257]: or `LOCAL_NETWORKS = ["katana", "devnet"]` and then check inclusion

**Review by Eikix — APPROVED** [review]
lgtm

File: src/kakarot/execution_context.cairo [hunk 2]
```diff
@@ -332,7 +332,8 @@ namespace ExecutionContext {
             keys_len=event.keys_len, keys=event.keys, data_len=event.data_len, data=event.data
         );
         // we maintain the semantics of one event struct involves iterating a full event struct size recursively
-        return emit_events(events_len - 1, events + 1 * model.Event.SIZE);
+        emit_events(events_len - 1, events + 1 * model.Event.SIZE);
+        return ();
```
Comment (jobez) [inline src/kakarot/execution_context.cairo:336]: was this causing subtle errors?

**Comment by jobez** [comment]
is the actual script that exports hints committed here?

**Review by jobez — APPROVED** [review]
ah! that explains it! other than my question of that change you made to event emission (not a blocker, can hear more at tomorrows standup) lgtm!

**Pull request merged**